qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -36,7 +39,60 @@ class CurveAnalysis(BaseCurveAnalysis):
     The fit parameters from the series defined under the analysis class are all shared
     and the analysis performs a single multi-objective function optimization.
 
-    See :class:`BaseCurveAnalysis` for overridable method documentation.
+    Subclass may override these methods to customize the fit workflow.
+
+    .. rubric:: _run_data_processing
+
+    This method performs data processing and returns the processed dataset.
+    By default, it internally calls :class:`DataProcessor` instance from the analysis options
+    and processes experiment data payload to create Y data with uncertainty.
+    X data and other metadata are generated within this method by inspecting the
+    circuit metadata. The series classification is also performed by based upon the
+    matching of circuit metadata and :attr:`SeriesDef.filter_kwargs`.
+
+    .. rubric:: _format_data
+
+    This method consumes the processed dataset and outputs the formatted dataset.
+    By default, this method takes the average of y values over
+    the same x values and then sort the entire data by x values.
+
+    .. rubric:: _generate_fit_guesses
+
+    This method creates initial guesses for the fit parameters.
+    This might be overridden by subclass.
+    See :ref:`curve_analysis_init_guess` for details.
+
+    .. rubric:: _run_curve_fit
+
+    This method performs the fitting with predefined fit models and the formatted dataset.
+    This method internally calls :meth:`_generate_fit_guesses` method.
+    Note that this is a core functionality of the :meth:`_run_analysis` method,
+    that creates fit result object from the formatted dataset.
+
+    .. rubric:: _evaluate_quality
+
+    This method evaluates the quality of the fit based on the fit result.
+    This returns "good" when reduced chi-squared is less than 3.0.
+    Usually it returns string "good" or "bad" according to the evaluation.
+    This criterion can be updated by subclass.
+
+    .. rubric:: _create_analysis_results
+
+    This method creates analysis results for important fit parameters
+    that might be defined by analysis options ``result_parameters``.
+
+    .. rubric:: _create_curve_data
```

Comment:
Good point. User can update the method to format data structure as they want. However, this will be removed once we switch to data frame because it doesn't need formatting and can be saved as-is.